src/ComposerJson.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace MartinsR\ComposerConstraintUpdater;
+
+class ComposerJson
+{
+    use FileOpener;
+    private string $composerContents;
+    /**
+     * @var array<string, array<string, string>>
+     */
+    private array $versionPrefixes;
+
+    /**
+     * @throws \Exception
+     */
+    public function __construct(private readonly string $composerPath, private readonly string $composerLockPath)
+    {
+        $this->composerContents = $this->fileContents($this->composerPath);
+        $this->versionPrefixes = $this->versionPrefixes();
+    }
+
+    /**
+     * @return array<mixed>
+     */
+    private function versionPrefixes(): array
+    {
+        $constraints = [];
+        foreach (['require', 'require-dev'] as $requirement) {
+            preg_match('/"' . preg_quote($requirement) . '"\s*:\s*\{([\s|\S]+?)\}/m', $this->composerContents, $matches);
+            if (isset($matches[1])) {
+                preg_match_all('/\"(.+?)\"\s*:\s*"([<>~^=]*).*"/m', $matches[1], $matches);
+                for ($i = 0; $i < count($matches[1]); $i++) {
+                    $constraints[$requirement][$matches[1][$i]] = $matches[2][$i] ?? '^';
+                }
+            }
+        }
+
+        return $constraints;
+    }
+
+    /**
+     * @param array<string, string> $packageConstraints
+     */
+    public function replaceVersions(array $packageConstraints): string
+    {
+        $requirements = ['require', 'require-dev'];
```

Comment:
small code duplication with versionPrefixes

Reply:
Maybe creating a class for traversing would be useful?
Something like:
` (new Traverser())->traverseComposerJson($composerJson, function(){
....
})`
And a different method for traversing composer lock.
But seems a bit too complicated.